ambari/datadog_checks/ambari/ambari.py

```diff
@@ -121,9 +121,16 @@ def get_service_status_and_metrics(
                     self.get_service_checks(base_url, cluster, service, service_tags)
 
     def get_service_checks(self, base_url, cluster, service, service_tags):
-        service_info = self._get_service_checks_info(base_url, cluster, service, service_tags)
-        for info in service_info:
-            self._submit_service_checks("state", info['state'], info['tags'])
+        service_check_endpoint = common.create_endpoint(base_url, cluster, service, SERVICE_INFO_QUERY)
+        service_resp = self._make_request(service_check_endpoint)
+        if service_resp is None:
+            self._submit_service_checks("state", self.CRITICAL, service_tags)
+            self.warning("No response received for service {}".format(service))
+        else:
+            state = service_resp['ServiceInfo']['state']
+            self._submit_service_checks(
+                "state", common.STATUS[state], service_tags + ['state:%s' % state], message=state
```

Comment:
Let's add a fallback mechanism for states like https://github.com/DataDog/integrations-core/blob/b9582a20eb80a4a9fc37938e8b99d6554f3ef642/ibm_db2/datadog_checks/ibm_db2/utils.py#L23-L24

Reply:
Good idea, done